word2gauss/embeddings.pyx

```diff
@@ -565,10 +595,20 @@ cdef class GaussianEmbedding:
             for pos_neg, fac in zip(target, [1.0, -1.0]):
                 for word_or_id in pos_neg:
                     if isinstance(word_or_id, basestring):
+                        # input is word, get its ID
                         word_id = vocab.word2id(word_or_id)
-                    else:
+                        mu_val = self.mu[word_id, :]
+                    elif isinstance(word_or_id, int):
+                        # input is ID
                         word_id = word_or_id
-                    t += fac * self.mu[word_id, :]
+                        mu_val = self.mu[word_id, :]
+                    # else:
+                    #     # input is a vector
+                    #     mu_val = word_or_id
+
+                    t += fac * mu_val
+        elif "numpy" in str(type(target)):
```

Comment:
Use the built in function `isinstance` here -- it's more precise and self documenting.  Otherwise I'm left guessing what the intention is here.